includes/Checker/Abstract_Check_Runner.php

```diff
@@ -243,16 +251,38 @@ private function get_shared_preparations( array $checks ) {
 	 * @since n.e.x.t
 	 *
 	 * @return array An array map of check slugs to Check instances.
+	 *
+	 * @throws Exception Thrown exception when a runtime check is requested and the plugin inactive.
 	 */
-	protected function get_checks_to_run() {
-		$check_slugs = $this->get_check_slugs();
-		$all_checks  = $this->get_checks_instance()->get_checks();
-
-		if ( empty( $check_slugs ) ) {
-			return $all_checks;
+	public function get_checks_to_run() {
```

Comment:
One thing that this method is missing (and FWIW the code has been missing before as well) is to check that the early initialization / runtime environment preparation is even possible. If for example the `object-cache.php` file cannot be placed, we won't be able to prepare the environment, in which case we also cannot do runtime checks.

So I think we should check here if the runner instance was _not_ initialized early, and if it wasn't, we need to exclude all runtime checks. In the case the check slugs were provided that include a runtime check, we should then also throw an exception to explain why those runtime checks cannot be run.

Reply:
Thanks @felixarntz I noticed this too. I've updated this method with the additional check.